File: Java/refactoring_exercise_4/src/test/java/za/co/entelect/refactoring4/controller/ImageServiceDelegateTest.java
package za.co.entelect.refactoring2.controller;

import org.junit.Test;
import za.co.entelect.refactoring4.controller.ImageServiceDelegate;
import za.co.entelect.refactoring4.domain.Image;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;

public class ImageServiceDelegateTest {

    private ImageServiceDelegate imageServiceDelegate = new ImageServiceDelegate();
    public static final String IMAGE_1 = "image_1";

    @Test
    public void testAddImage(){
        assertEquals(1, imageServiceDelegate.count());
        imageServiceDelegate.add(new Image(IMAGE_1, new byte[]{}));
        assertEquals(2, imageServiceDelegate.count());
    }

    @Test
    public void testFetchImage(){
        assertNull(imageServiceDelegate.fetch(IMAGE_1));
    }
}

File: C#/Refactoring/refactoring_exercise_2/test/za/co/entelect/refactoring2/controller/ImageServiceDelegateTest.cs
using System;
using NUnit.Framework;
using NUnit.Framework.Constraints;
using refactoring_exercise_2.za.co.entelect.refactoring2.controller;
using refactoring_exercise_2.za.co.entelect.refactoring2.domain;
using refactoring_exercise_2.za.co.entelect.refactoring2.service;

namespace refactoring_exercise_2.test.za.co.entelect.refactoring2.controller
{
    [TestFixture]
    public class ImageServiceDelegateTest {

        private readonly ImageServiceDelegate _imageServiceDelegate = new ImageServiceDelegate();
        public const String Image1 = "image_101";

        [Test]
        public void TestAddImage(){
            Assert.That(_imageServiceDelegate.Count(), new EqualConstraint(1));
            _imageServiceDelegate.Add(new Image(Image1, new byte[]{}));
            Assert.That(_imageServiceDelegate.Count(), new EqualConstraint(2));
        }

        [Test]
        public void TestFetchImage(){
            var actual = _imageServiceDelegate.Fetch(ImageService.DefaultImage);
            Assert.NotNull(actual);
        }
    }
}